### PR TITLE
XRT-620: xbutil changes for aie reset, program, query

### PR DESCRIPTION
**Pull request by Himanshu-xilinx**
- implemented xbutil reset to reset aie array
- xbutil program to load aie pdi
- created sysfs node to expose aie metadata
- xbutil query to show aie_metadata also

### CODE REVIEW

File: src/runtime_src/core/edge/tools/xbutil/CMakeLists.txt [hunk 1]
```diff
@@ -10,15 +10,31 @@ file(GLOB XRT_EDGE_TOOLS_XBUTIL_FILES
...
-  xrt_core_static
-  xrt_coreutil_static
-  ${Boost_FILESYSTEM_LIBRARY}
-  ${Boost_SYSTEM_LIBRARY}
-  pthread
-  rt
-  uuid
-  ${CURSES_LIBRARIES}
-  )
+if (DEFINED XRT_AIE_BUILD)
```
Comment (rozumx) [inline src/runtime_src/core/edge/tools/xbutil/CMakeLists.txt:13]: Would it be better to simply leave the existing target_link_libraries alone and add a new AIE library.  For example:
```
if (DEFINED XRT_AIE_BUILD)
  target_link_libraries(xbutil xaiengine)
endif()
```

File: src/runtime_src/core/edge/tools/xbutil/xbutil.h [hunk 2]
```diff
@@ -293,6 +298,23 @@ class device {
         }
     }
 
+#ifdef XRT_AIE_BUILD
+    void getAieMetadata() const {
+        std::string errmsg;
```
Comment (rozumx) [inline src/runtime_src/core/edge/tools/xbutil/xbutil.h:303]: Shouldn't the error message be checked for any errors.

File: src/runtime_src/core/edge/tools/xbutil/xbutil.h [hunk 3]
```diff
@@ -293,6 +298,23 @@ class device {
...
+        std::vector<char> buf, temp_buf;
```
Comment (rozumx) [inline src/runtime_src/core/edge/tools/xbutil/xbutil.h:304]: The variables don't seemed to be used.

File: src/runtime_src/core/edge/tools/xbutil/xbutil.h [hunk 4]
```diff
@@ -293,6 +298,23 @@ class device {
...
+        std::vector<std::string> aie_buf, stream_stat;
```
Comment (rozumx) [inline src/runtime_src/core/edge/tools/xbutil/xbutil.h:305]: stream_stat doesn't seem to be used.

**Comment by chvamshi-xilinx** [comment]
Closing as changes in xbutil2 is ready for this feature

**Pull request closed without merging**